### ISSUE TITLE
Provider-Issue: base64 error on get_short_epg 

### ISSUE THREAD

**Issue by kingdase** [pierre-emmanuelJ/iptv-proxy#115]
`[GIN-debug] Listening and serving HTTP on :8080
2022/03/08 09:48:01 [iptv-proxy] 2022/03/08 - 09:48:01 | 192.168.178.29 |Action
[GIN] 2022/03/08 - 09:48:01 | 200 |    129.3953ms |  192.168.178.29 | GET      "/player_api.php?username=usertest&password=passwordtest"
2022/03/08 09:48:13 [iptv-proxy] 2022/03/08 - 09:48:13 | 192.168.178.29 |Action get_live_categories
[GIN] 2022/03/08 - 09:48:13 | 200 |     188.348ms |  192.168.178.29 | GET      "/player_api.php?username=usertest&password=passwordtest&action=get_live_categories"
2022/03/08 09:48:25 [iptv-proxy] 2022/03/08 - 09:48:25 | 192.168.178.29 |Action get_live_streams
[GIN] 2022/03/08 - 09:48:25 | 200 |    214.6709ms |  192.168.178.29 | GET      "/player_api.php?username=usertest&password=passwordtest&action=get_live_streams&category_id=807"
2022/03/08 09:48:25 [iptv-proxy] 2022/03/08 - 09:48:25 | 192.168.178.29 |Action get_short_epg
[GIN] 2022/03/08 - 09:48:25 | 200 |    293.8188ms |  192.168.178.29 | GET      "/player_api.php?username=usertest&password=passwordtest&action=get_short_epg&stream_id=438692"
[GIN] 2022/03/08 - 09:48:38 | 200 |    173.9524ms |  192.168.178.29 | GET      "/player_api.php?username=usertest&password=passwordtest&action=get_short_epg&stream_id=438620"
Error #01: illegal base64 data at input byte 19
[GIN] 2022/03/08 - 09:48:40 | 200 |    230.5584ms |  192.168.178.29 | GET      "/player_api.php?username=usertest&password=passwordtest&action=get_short_epg&stream_id=438620"
Error #01: illegal base64 data at input byte 19
[GIN] 2022/03/08 - 09:48:47 | 200 |     175.119ms |  192.168.178.29 | GET      "/player_api.php?username=usertest&password=passwordtest&action=get_short_epg&stream_id=438620"
Error #01: illegal base64 data at input byte 19`

**Comment by Olivier6767** [comment]
I confirm I have the same issue with get_short_epg.
The same error also happens with get_simple_data_table.

**Comment by kingdase** [comment]
The base64 decode problem comes from https://github.com/tellytv/go.xtream-codes/blob/master/base64.go
Orignal from https://github.com/manifoldco/go-base64/blob/master/base64.go

**Why does the EPG data has to be processed at all?**

As a proxy, it should simply be passed through without processing?

You can verify with 
```
UnmarshalJSON ( []byte("\"WkRGLU1vcmdlbm1hZ2F6aW4=\"") )
```

Source code excerpt from go.xtream-codes/blob/master/base64.go:

```
func (bv *Base64Value) UnmarshalJSON(b []byte) error {
	if len(b) < 2 || b[0] != byte('"') || b[len(b)-1] != byte('"') {
		return errors.New("value is not a string")
	}

	out := make([]byte, base64.RawURLEncoding.DecodedLen(len(b)-2))
	n, err := base64.RawURLEncoding.Decode(out, b[1:len(b)-1])
	if err != nil {
		return err
	}

	v := reflect.ValueOf(bv).Elem()
	v.SetBytes(out[:n])
	return nil
}
```

**Comment by kingdase** [comment]
RawURLEncoding is without padding.
So remove the '=' in my example and it works.
For a workaround, you can replace _RawURLEncoding_ with _StdEncoding_ in UnmarshalJSON.

```
UnmarshalJSON ( []byte("\"WkRGLU1vcmdlbm1hZ2F6aW4\"") )
```

**Comment by pierre-emmanuelJ** [comment]
Yes I need also to make a patch on the base64 stuff in go.xtream-code when I'm less busy, I think I'll  fork the  go.xtream-code repo to be quicker...etc

Thanks all for reporting and debugging this issue :)

**Comment by xxn4** [comment]
I got my EPG running with these changes:
https://github.com/tellytv/go.xtream-codes/compare/master...xxn4:go.xtream-codes:master

I don't think that especially suppressing the error message in line 54 of base64.go is the best way but at least EPG works now. Maybe it helps someone.
